clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -1100,6 +1101,8 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
       // Rerun aspect propagation without warning diagnostics.
       MPM.addPass(SYCLPropagateAspectsUsagePass(/*ExcludeAspects=*/{},
                                                 /*ValidateAspects=*/false));
+      // Remove SYCL metadata added by sycl-post-link tool, like sycl_aspects
+      MPM.addPass(CleanupSYCLCompilerInternalMetadataPass());
```

Comment:
Could we do it after `CompileTimePropertiesPass` in case we want to clean something after that too? I could see that happening potentially.